sdk/keyvault/keyvault-keys/src/keysModels.ts

```diff
@@ -268,3 +268,43 @@ export interface RequestOptions {
    */
   requestOptions?: msRest.RequestOptionsBase;
 }
+
+/**
+ * @interface
+ * An interface that tracks the settings for paged iteration
+ */
+export interface PageSettings {
+  /**
+   * @member {string} [continuationToken] The token that keeps track of where to continue the iterator
+   */
+  continuationToken?: string,
+  /**
+   * @member {number} [pageSize] The size of the page during paged iteration
+   */
+  pageSize?: number,
+}
+
+/**
+ * @interface
+ * An interface that allows async iterable iteration both to completion and by page.
+ */
+export interface PagedAsyncIterableIterator<T> {
```

Comment:
I like this type. Should it live someplace more common? Storage maybe wants this type too... /cc @HarshaNalluru

Reply:
Thanks, @bterlson. 
I like it too. I'll add it for storage.

Reply:
I've added a new @azure/core-paging that we can share.